### PR TITLE
Add ags_log endpoint

### PR DESCRIPTION
**Pull request by KoalaGeo**
Requests synthetic PDF log from Groundhog Web Viewer API (https://webservices.bgs.ac.uk/GWBV/viewborehole?loca_id=20200204094811691620) using BGS Loca ID. 

Currently gets a 200 response but PDF doesn't display. Tried as both `inline` & `attachment` but PDF files are only 1kb - it's not actually pulling the PDF from the web viewer. 

Addresses https://github.com/BritishGeologicalSurvey/pyagsapi/issues/100

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -179,3 +181,14 @@ def prepare_validation_response(request, data):
         'data': data,
     }
     return ValidationResponse(**response_data, media_type="application/json")
+
+
+@router.get("/ags_log/{bgs_loca_id}")
```
Comment (ximenesuk) [inline app/routes.py:186]: This decorator needs to specify the `response_class` and `responses` for the openapi. You'll need top create a `pdf_responses` dictionary - see the other routes and the top of the file for examples.

You should also either use or create an error class (see `errors.py`) for a full error response.

File: Dockerfile [hunk 2]
```diff
@@ -4,6 +4,5 @@ FROM tiangolo/uvicorn-gunicorn-fastapi:python3.11-slim-2023-02-20
 COPY requirements.txt .
 RUN pip install -r requirements.txt
 
-
 RUN rm -rf /app/*
-COPY ./app /app/app
+COPY ./app /app/app
```
Comment (ximenesuk) [inline Dockerfile:8]: You should put back the new line at the end of the file.

**Comment by ximenesuk** [comment]
It would be good to add a basic integration test too.

File: app/routes.py (hunk 1 above)
Comment (KoalaGeo) [reply app/routes.py:190]: Started in https://github.com/BritishGeologicalSurvey/pyagsapi/pull/101/commits/be7ca46b37eaffcbab8c64e6153ed8fae6ad9027

Comment (KoalaGeo) [reply app/routes.py]: It's not using the `errors.py` as expected, please review

**Comment by KoalaGeo** [comment]
It's working approximately as required, outstanding:

- Change output file name to = submitted BGS_LOCA_ID
- Make it use `errors.py` in response
  - doesn't need a new response, the existing one in `errors.py` is appropriate
- Add integration tests

**Comment by KoalaGeo** [comment]
> It would be good to add a basic integration test too.

Started with https://github.com/BritishGeologicalSurvey/pyagsapi/pull/101/commits/47e4298841286283d09da543d8398bb08d0606b5 but getting error `TypeError: 'function' object does not support the asynchronous context manager protocol`

**Comment by KoalaGeo** [comment]
Working with exception of error reporting:

Need to specify `response_class`  & make it use errors.py in response. The existing errors in errors.py is appropriate.

File: test/integration/test_api.py [hunk 3]
```diff
@@ -408,6 +408,60 @@ async def test_validate_dictionary_choice(async_client, dictionary, filename, ex
...
+    assert response.headers["Content-Type"] == "application/pdf"
+    # Check that the response content is not empty
+    assert len(response.content) > 0
+    # Check it is a PDF file
+    assert response.content.startswith(b'%PDF')
+
+
+def test_get_unknown_ags_log(client):
+    """
+    Confirm that the endpoint can return the expected .pdf.
```
Comment (ximenesuk) [inline test/integration/test_api.py:447]: Darn, this comment needs updating!

Comment (KoalaGeo) [reply test/integration/test_api.py]: https://github.com/BritishGeologicalSurvey/pyagsapi/pull/101/commits/7d482295a5467af05523265942ac9c1164ee13c1 fixed

**Review by KoalaGeo — COMMENTED** [review]
Need to add openapi_tags=tags_metadata or equivalent @ximenesuk

**Pull request merged**